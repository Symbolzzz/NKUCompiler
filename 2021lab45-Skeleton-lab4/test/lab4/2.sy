// i am a function: this function has many syntax errors.

int f(){
	int a;
	a = 0;
	while(a<10){
		a = a * 2345;
	}
	for(int a=0;i<10;i++)
	a++;
	if 
	else

	return a;
}

int main(){
	int _testid;
	_testid = 0;
	if(_testid==0){
		int _testid;
		int _testid2;
		_testid1=_testid1+1;
	}
	/*
		Comment lines;
	*/
	{
		int a;
	}
	{
		int a;
	}
	int b;
	return 0;
}

/*
	Comment lines.
*/
