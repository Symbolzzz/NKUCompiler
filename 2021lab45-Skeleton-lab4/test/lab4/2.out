token	lexeme	lineno
INT	int	3
 ID	f	3
LPAREN	(	3
RPAREN	)	3
LBRACE	{	3
INT	int	4
 ID	a	4
SEMICOLON	;	4
ID	a	5
 ASSIGN	=	5
 NUMBER	0	5
SEMICOLON	;	5
WHILE	while	6
LPAREN	(	6
ID	a	6
LESS	<	6
NUMBER	10	6
RPAREN	)	6
LBRACE	{	6
ID	a	7
 ASSIGN	=	7
 ID	a	7
 MUL	*	7
 NUMBER	2345	7
SEMICOLON	;	7
RBRACE	}	8
RETURN	return	9
 ID	a	9
SEMICOLON	;	9
RBRACE	}	10
INT	int	12
 ID	main	12
LPAREN	(	12
RPAREN	)	12
LBRACE	{	12
INT	int	13
 ID	_testid	13
SEMICOLON	;	13
ID	_testid	14
 ASSIGN	=	14
 NUMBER	0	14
SEMICOLON	;	14
IF	if	15
LPAREN	(	15
ID	_testid	15
EQUAL	==	15
NUMBER	0	15
RPAREN	)	15
LBRACE	{	15
INT	int	16
 ID	_testid	16
SEMICOLON	;	16
INT	int	17
 ID	_testid2	17
SEMICOLON	;	17
ID	_testid1	18
ASSIGN	=	18
ID	_testid1	18
ADD	+	18
NUMBER	1	18
SEMICOLON	;	18
RBRACE	}	19
LBRACE	{	21
INT	int	22
 ID	a	22
SEMICOLON	;	22
RBRACE	}	23
LBRACE	{	24
INT	int	25
 ID	a	25
SEMICOLON	;	25
RBRACE	}	26
INT	int	27
 ID	b	27
SEMICOLON	;	27
RETURN	return	28
 NUMBER	0	28
SEMICOLON	;	28
RBRACE	}	29
